src/main/java/hu/blackbelt/structured/map/proxy/MapProxy.java

```diff
@@ -706,6 +708,80 @@ private boolean invokeEquals(Object proxy, Object[] args) throws InvocationTarge
         return false;
     }
 
+    private void invokeAdd(Method m, Object[] args) throws ExecutionException {
+        if (params.isImmutable()) {
+            throw new IllegalStateException("Could not call set on immutable object");
```

Comment:
> Could not call set on immutable object

Shouldn't this be "Add cannot be called on immutable object" ?

Reply:
b7d7975